rmw_connextdds_common/src/ndds/dds_api_ndds.cpp

```diff
@@ -215,44 +225,42 @@ rmw_connextdds_initialize_participant_qos_impl(
         // https://community.rti.com/static/documentation/connext-dds/6.1.1/doc/manuals/connext_dds_professional/properties_reference/index.html
         if (ctx->discovery_options) {
           const auto range = ctx->discovery_options->automatic_discovery_range;
+          bool set_multicast_peers = false;
           switch (range) {
             case RMW_AUTOMATIC_DISCOVERY_RANGE_SUBNET:
               /* No action needed. This is the default discovery behavior for DDS */
               break;
+            case RMW_AUTOMATIC_DISCOVERY_RANGE_DEFAULT:
+            case RMW_AUTOMATIC_DISCOVERY_RANGE_LOCALHOST:
+            case RMW_AUTOMATIC_DISCOVERY_RANGE_OFF:
+              set_multicast_peers = true;
+              break;
             default:
```

Comment:
Thinking a bit about this, why should we accept an invalid range type?

Is it to support future values? because if so, they wouldn't really be "supported", just "accepted" and treated like another value, but they most likely won't behave as expected.

I feel like it would be better to return an error so that a user knows the RMW will not behave as expected with the selected mode.

Reply:
We shouldn't, thanks for pointing this out. We are updating the other implementations too, and I cherry-picked your changes to make this the case in connext too.